bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/service/SharingStateService.kt

```diff
@@ -93,6 +90,88 @@ class SharingStateService(private val stateRepository: SharingStateRepository) {
                 sharingProcessStarted = it.sharingProcessStarted
             )
         })
+    }
+
+    fun setInitial(externalIds: Collection<String>, businessPartnerType: BusinessPartnerType): Collection<SharingState> {
+        val sharingStates = getOrCreate(externalIds, businessPartnerType)
+        return sharingStates.map { setInitial(it) }
+    }
+
+    fun setSuccess(externalIdBpnDto: Collection<ExternalIdBpnDto>, businessPartnerType: BusinessPartnerType): Collection<SharingState> {
+        val sharingStates = getOrCreate(externalIdBpnDto.map { it.externalId }, businessPartnerType)
+        val externalIdByBpn = externalIdBpnDto.associateBy { it.externalId }
```

Comment:
Variable name is confusing.
```suggestion
        val bpnByExternalId = externalIdBpnDto.associateBy { it.externalId }.mapValues { it.value.bpn }
```

Reply:
Yep, makes sense